File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

#include <sstream>
#include <string>

namespace raindrops
{
std::string convert(int num)
{
    std::stringstream result;
    if (num % 3 == 0)
    {
        result << "Pling";
    }
    if (num % 5 == 0)
    {
        result << "Plang";
    }
    if (num % 7 == 0)
    {
        result << "Plong";
    }
    if (num % 3 != 0 and num % 5 != 0 and num % 7 != 0)
    {
        result << std::to_string(num);
    }

    return result.str();
}
} // namespace raindrops

File: zig/raindrops/raindrops.zig
const std = @import("std");

pub fn convert(buffer: []u8, n: u32) []const u8 {
    var pos: usize = 0;
    if (n % 3 == 0) {
        @memcpy(buffer.ptr + pos, "Pling");
        pos += 5;
    }
    if (n % 5 == 0) {
        @memcpy(buffer.ptr + pos, "Plang");
        pos += 5;
    }
    if (n % 7 == 0) {
        @memcpy(buffer.ptr + pos, "Plong");
        pos += 5;
    }
    if (pos == 0) {
        pos += std.fmt.printInt(buffer, n, 10, std.fmt.Case.lower, .{});
    }
    
    return buffer[0..pos];
}
